ADR/ADR021-use-auth0.md

```diff
@@ -0,0 +1,41 @@
+# ADR021: Use Auth0 as an Identity Provider for Authentication
+
+Date: 2023-08-09
+
+## Status
+
+Accepted
+
+## Context
+
+Following [ADR015: Use an Identity Provider for authentication and implement authorisation](ADR015-idp-authentication.md) we looked into alternatives to GOV.UK Signon.
+
+[NCSC guidance on Enterprise authentication policy](https://www.ncsc.gov.uk/collection/device-security-guidance/infrastructure/enterprise-authentication-policy) states:
+
+> Where possible, reduce reliance on passwords and implement passwordless authentication
+
+and there is [NCSC guidance on one time passwords](https://www.ncsc.gov.uk/guidance/authentication-methods-choosing-the-right-type#section_5).
+
+[UK Government Security Single Sign-On](https://sso.service.security.gov.uk/) was considered, but is still in an "Alpha" phase.
+
+[Auth0 offers Passwordless Authentication Methods](https://auth0.com/docs/authenticate/passwordless/authentication-methods) so was also considered, and a DPIA (Data Protection Impact Assessment) was carried out.
+
+[User research on authentication](../research/2023-08-Usability_Testing_Authentication.md) showed that participants were comfortable using a passwordless process and preferred the Auth0 journey.
```

Comment:
Is it worth calling out that Auth0 can't be configured to use the design system exactly (from what I recall)? I don't think this was actually picked up on by any one in the user research but it was a concern we had beforehand.

Reply:
Thanks @jamie-o-wilkinson , good suggestion. I'll add something to the "Consequences" section, but if you're able to reference any specific concerns they could be included.